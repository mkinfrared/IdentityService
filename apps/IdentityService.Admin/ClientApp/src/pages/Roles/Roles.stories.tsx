import { Meta, StoryObj } from "@storybook/react";

import Roles from "./Roles";
import css from "./Story.module.scss";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "UI/Roles",
  component: Roles,
  tags: ["autodocs"],
} satisfies Meta<typeof Roles>;

const Default: Story = {
  args: {},
  render: (args) => (
    <div className={css.Story}>
      <Roles {...args} />
    </div>
  ),
};

export { Default };

export default meta;
