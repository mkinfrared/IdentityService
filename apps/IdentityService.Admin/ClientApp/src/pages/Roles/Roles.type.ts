export type RolesProps = {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
};
