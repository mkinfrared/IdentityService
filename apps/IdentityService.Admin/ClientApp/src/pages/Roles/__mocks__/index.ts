export { default } from "./Roles.mock";
