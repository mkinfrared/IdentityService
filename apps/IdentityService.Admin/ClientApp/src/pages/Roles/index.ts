export { default } from "./Roles";

export * from "./Roles.type";
