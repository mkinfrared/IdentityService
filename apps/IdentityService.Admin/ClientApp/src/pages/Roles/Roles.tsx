import axios from "axios";
import { useEffect } from "react";

import css from "./Roles.module.scss";
import { RolesProps } from "./Roles.type";

const Roles = ({ className }: RolesProps) => {
  const classNames = [css.Roles, className].filter(Boolean).join(" ");

  // https://jsonplaceholder.typicode.com/todos/1?list[]=foo&list[]=bar&list[]=dash&list[]=dot
  useEffect(() => {
    axios.get("https://jsonplaceholder.typicode.com/todos/1", {
      params: {
        list: ["foo", "bar", "dash", "dot"],
      },
    });
  }, []);

  return (
    <div className={classNames} data-testid="Roles">
      Roles works!!
    </div>
  );
};

export default Roles;
