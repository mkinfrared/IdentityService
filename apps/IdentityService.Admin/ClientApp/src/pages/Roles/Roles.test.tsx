import { render } from "@testing-library/react";

import Roles from "./Roles";

describe("<Roles />", () => {
  const Component = <Roles />;

  it("should be defined", () => {
    expect(Roles).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("Roles");

    expect(element).toBeDefined();
  });
});
