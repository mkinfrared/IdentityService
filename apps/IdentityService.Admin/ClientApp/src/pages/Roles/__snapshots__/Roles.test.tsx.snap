// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Roles /> should match the snapshot 1`] = `
<div>
  <div
    class="Roles"
    data-testid="Roles"
  >
    Roles works!!
  </div>
</div>
`;
